src/Utils/messages.ts

```diff
@@ -868,12 +869,24 @@ const REUPLOAD_REQUIRED_STATUS = [410, 404]
 /**
  * Downloads the given message. Throws an error if it's not a media message
  */
-export const downloadMediaMessage = async(
```

Comment:
can be simplified

```ts
export async function downloadMediaMessage(
  message: WAMessage,
  type: 'buffer' | 'stream',
  options: MediaDownloadOptions,
  ctx?: DownloadMediaMessageContext
): Promise<Buffer | Transform> {
  try {
    const result = await downloadMsg()
    return result
  } catch(error) {
    if(ctx && axios.isAxiosError(error) && REUPLOAD_REQUIRED_STATUS.includes(error.response?.status!)) {
      ctx.logger.info({ key: message.key }, 'sending reupload media request...')
      message = await ctx.reuploadRequest(message)
      return downloadMsg()
    }
    throw error
  }
```

Reply:
i'm sorry. i don't think so. in the way i've typed, the return type depends on the parameters. in that way:
- if `type = 'buffer'`, returns `Promise<Buffer>`
- if `type = 'stream'`, returns `Promise<Transform>`

Reply:
```ts
export async function downloadMediaMessage<T extends 'buffer' | 'stream'>(
    message: WAMessage,
    type: T,
    options: MediaDownloadOptions,
    ctx?: DownloadMediaMessageContext
): Promise<T extends 'buffer' ? Buffer : Transform> {
   //
}
```

Reply:
good idea to use extends! thank you
i wasn't able to make it exactly in that way (_typescript typing gymnastics_), but it's simpler than before